packages/hub/src/lib/parse-safetensors-metadata.ts

```diff
@@ -11,6 +11,9 @@ import { promisesQueue } from "../utils/promisesQueue";
 const SINGLE_FILE = "model.safetensors";
 const INDEX_FILE = "model.safetensors.index.json";
 const PARALLEL_DOWNLOADS = 5;
+const MAX_HEADER_LENGTH = 25_000_000;
```

Comment:
No idea if this value (25MB) is reasonable

Reply:
i think 25MB is more than enough for the JSON part!! That's one big JSON already.

cc @Narsil for visibility (we had a pathologically crafted safetensors file that crashed the Hub) – probably we don't need to put that in the spec

Reply:
https://github.com/huggingface/safetensors/blob/main/safetensors/src/tensor.rs#L295

https://github.com/huggingface/safetensors/blob/main/safetensors/src/tensor.rs#L10

There's  a hard cap in the lib to prevent DOS attack (while deserializing super large JSON structures)